src/workers/runners/AbstractRunner.ts

```
@@ -1,51 +1,83 @@
 import { SYNC_FILENAMES, UpdateSyncFile, WaitUntilDone } from '../../utils/Sync';
 import { sleep } from '../../utils/Utils';
-import { BaseWorker } from '../BaseWorker';
-import { EmptyConfiguration, WorkerConfiguration } from '../configuration/WorkerConfiguration';
+import { EmptyConfiguration } from '../configuration/WorkerConfiguration';
+import { RunWorkable } from './interfaces/RunWorkable';
+import { Runnable } from './interfaces/Runnable';
 
-export class AbstractRunner {
+export abstract class AbstractRunner implements Runnable {
   static RUN_EVERY_MINUTES = 60;
 
-  fetchersToLaunch: BaseWorker<WorkerConfiguration>[];
+  workersToLaunch: RunWorkable[];
+  mutex;
+  shouldWait;
 
-  constructor(toLaunch: BaseWorker<WorkerConfiguration>[]) {
-    this.fetchersToLaunch = toLaunch;
+  async init(): Promise<void> {
+    for (const fetcherToLaunch of this.workersToLaunch) {
+      try {
+        await fetcherToLaunch.init();
+      } catch (e) {
+        console.warn('Could not load configuration for worker ' + fetcherToLaunch.monitoringName + '. Skipping it.');
```

I think it should throw an error and exit instead of silently failing